invenio_vocabularies/services/config.py

```diff
@@ -0,0 +1,205 @@
+        fields=[
+            "id.text^100",
+            "id.text._2gram",
+            "id.text._3gram",
+            "title.en^5",
+            "title.en._2gram",
+            "title.en._3gram",
+        ],
+    )
+
+    sort_default = "bestmatch"
+
+    sort_default_no_query = "title"
+
+    sort_options = {
+        "bestmatch": dict(
+            title=_("Best match"),
+            fields=["_score"],  # ES defaults to desc on `_score` field
+        ),
+        "title": dict(
+            title=_("Title"),
+            fields=["title_sort"],
+        ),
+        "newest": dict(
+            title=_("Newest"),
+            fields=["-created"],
+        ),
+        "oldest": dict(
+            title=_("Oldest"),
+            fields=["created"],
+        ),
+    }
+
+
+class VocabularyTypeSearchOptions(SearchOptions):
+    """Search options for vocabulary types."""
+
+    # TODO: Is this still necessary here?
+    params_interpreters_cls = [
+        FilterParam.factory(param="tags", field="tags"),
+    ] + SearchOptions.params_interpreters_cls
+
+    # TODO: Is this still necessary here?
+    suggest_parser_cls = SuggestQueryParser.factory(
+        fields=[
+            "id.text^100",
+            "id.text._2gram",
+            "id.text._3gram",
+        ],
+    )
+
+    sort_options = {
+        "id": dict(
+            title=_("ID"),
+            fields=["id"],
+        ),
+    }
+
+    sort_default = "id"
+
+    sort_default_no_query = "id"
+
+    sort_direction_options = {
+        "asc": dict(title=_("Ascending"), fn=asc),
+        "desc": dict(title=_("Descending"), fn=desc),
+    }
+
+    sort_direction_default = "asc"
+
+
+class VocabulariesServiceConfig(RecordServiceConfig):
+    """Vocabulary service configuration."""
+
+    service_id = "vocabularies"
+    indexer_queue_name = "vocabularies"
+    permission_policy_cls = PermissionPolicy
+    record_cls = Vocabulary
+    schema = VocabularySchema
+    task_schema = TaskSchema
+
+    search = VocabularySearchOptions
+
+    components = [
+        # Order of components are important!
+        VocabularyTypeComponent,
+        DataComponent,
+        PIDComponent,
+    ]
+
+    links_item = {
+        "self": Link(
+            "{+api}/vocabularies/{type}/{id}",
+            vars=lambda record, vars: vars.update(
+                {
+                    "id": record.pid.pid_value,
+                    "type": record.type.id,
+                }
+            ),
+        ),
+    }
+
+    links_search = pagination_links("{+api}/vocabularies/{type}{?args*}")
+
+
+class VocabularyTypesServiceConfig(RecordServiceConfig):
+    """Vocabulary types service configuration."""
+
+    service_id = "vocabulary_types"
+    permission_policy_cls = PermissionPolicy
+    record_cls = Vocabulary  # TODO: Is this correct?
+    schema = VocabularySchema
+    task_schema = TaskSchema
+    vocabularies_listing_resultlist_cls = results.VocabularyMetadataList
+
+    vocabularies_listing_item = {
```

Comment:
```suggestion
    result_list_cls = results.VocabularyMetadataList

    links_item = {
```

minor: since it's the only type of result we have in the service

Reply:
There is actually another links_item below - how should I treat this?